src/rationals.rs

```diff
@@ -0,0 +1,245 @@
+use std::fmt::Formatter;
+use super::latex::LaTeXable;
+
+/// This struct represents the rational number
+/// `n/d` where `n` is the numerator and `d` is the denominator.
+/// The sign of the rational number is always stored in the numerator.
+#[derive(Debug)]
+struct Rational {
+    numerator: i64,
+    denominator: u64,
+}
+
+impl Rational {
+    fn normalize(&mut self) {
+        let gcd = gcd(self.numerator.unsigned_abs(), self.denominator);
+        self.numerator /= gcd as i64;
+        self.denominator /= gcd;
+    }
+
+    fn new_non_signed_denominator(numerator: i64, denominator: u64) -> Rational {
+        let mut r = Rational { numerator, denominator };
+        r.normalize();
+        r
+    }
+
+    pub fn new(numerator: i64, denominator: i64) -> Rational {
+        let sign = denominator.signum();
+        let mut r = Rational { numerator: numerator * sign, denominator: denominator.unsigned_abs() };
+        r.normalize();
+        r
+    }
+
+    pub fn from_integer(n: i64) -> Rational {
+        Rational { numerator: n, denominator: 1 }
+    }
+
+    fn cmp(&self, other: &Rational) -> std::cmp::Ordering {
+        let a = self.numerator * other.denominator as i64;
+        let b = other.numerator * self.denominator as i64;
+        a.cmp(&b)
+    }
+}
+
+impl LaTeXable for Rational {
+    fn to_latex(&self) -> String {
+        let sign = if self.numerator < 0 { "-" } else { "" };
+        match self.denominator {
+            1 => format!("{}", self.numerator),
+            _ => format!("{}\\frac{{{}}}{{{}}}", sign, self.numerator.unsigned_abs(), self.denominator),
+        }
+    }
+}
+
+impl std::fmt::Display for Rational {
+    fn fmt(&self, f: &mut Formatter) -> std::fmt::Result {
+        match self.denominator {
+            1 => write!(f, "{}", self.numerator),
+            _ => write!(f, "{}/{}", self.numerator, self.denominator),
+        }
+    }
+}
+
+impl std::ops::Add for Rational {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self::new_non_signed_denominator(
+            self.numerator * other.denominator as i64 + other.numerator * self.denominator as i64,
+            self.denominator * other.denominator
+        )
+    }
+}
+
+impl std::ops::Neg for Rational {
+    type Output = Self;
+
+    fn neg(self) -> Self {
+        Self::new_non_signed_denominator(-self.numerator, self.denominator)
+    }
+}
+
+impl std::ops::Sub for Rational {
+    type Output = Self;
+
+    fn sub(self, other: Self) -> Self {
+        self + -other
+    }
+}
+
+impl std::ops::Mul for Rational {
+    type Output = Self;
+
+    fn mul(self, other: Self) -> Self {
```

Comment:
I'd use checked_mul and return Option<Self> (and same for other ops). Overflows aren't even checked in release build, I think it'd better to somehow notify the user and stop the calculations.